CircleCI should install and run the nighty versoin of rust

I'm getting this in compiling parity's code:

```
error[E0554]: #![feature] may not be used on the stable release channel
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hash256-std-hasher-0.12.4/src/lib.rs:16:35
   |
16 | #![cfg_attr(not(feature = "std"), feature(core_intrinsics))]
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0554`.

```
for which rustc says
```
root@54d6870151e0:~/project# rustc --explain E0554
Feature attributes are only allowed on the nightly release channel. Stable or
beta compilers will not comply.

Example of erroneous code (on a stable compiler):

#![feature(non_ascii_idents)] // error: #![feature] may not be used on the
                              //        stable release channel

If you need the feature, make sure to use a nightly release of the compiler
(but be warned that the feature may be removed or altered in the future).
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2
 jobs:
   runTests:
     docker:
-      - image: rust:1.33-slim
+      - image: rustlang/rust:nightly-slim
     steps:
       - checkout
       - run:

